File: typescript-client-sdk/src/sdk/models/operations/getonecrmaccount.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetOneCrmAccountQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=id" })
  id: string;
}


export class GetOneCrmAccountResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=account" })
  account?: shared.Account;
}


export class GetOneCrmAccountRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetOneCrmAccountQueryParams;
}


export class GetOneCrmAccountResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetOneCrmAccountResponseBody;
}

File: go-client-sdk/pkg/models/operations/getonecrmaccount.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetOneCrmAccountQueryParams struct {
	AccessToken string `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool  `queryParam:"style=form,explode=true,name=allFields"`
	ID          string `queryParam:"style=form,explode=true,name=id"`
}

type GetOneCrmAccountResponseBody struct {
	Account *shared.Account `json:"account,omitempty"`
}

type GetOneCrmAccountRequest struct {
	QueryParams GetOneCrmAccountQueryParams
}

type GetOneCrmAccountResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetOneCrmAccountResponseBody
}
